src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -823,7 +833,7 @@ static int validateAndParseBoolean(String columnName, Object input, long insertR
 
   static void checkValueInRange(
```

Comment:
Although unlikely might want to add a `@throws IndexOutOfBoundsException` if `precision-scale` exceeds `Power10.sb16Size`

Reply:
Good call. It made me realize that I was changing the behavior a bit. I have now changed it to fallback to old behavior if we get unexpected input.